radixdlt/src/integration/java/com/radixdlt/integration/distributed/simulation/network/SimulationNetwork.java

```diff
@@ -113,51 +120,21 @@ public String toString() {
 		}
 	}
 
-	/**
-	 * The latency configuration for a network
-	 */
-	public interface LatencyProvider {
-
-		/**
-		 * If >= 0, returns the latency in milliseconds of the next message.
-		 * If < 0, signifies to drop the next message.
-		 *
-		 * @param msg the next message
-		 * @return the latency in milliseconds if >= 0, otherwise a negative number signifies a drop
-		 */
-		int nextLatency(MessageInTransit msg);
+	public interface ChannelCommunication {
+		Observable<MessageInTransit> transform(BFTNode sender, BFTNode receiver, Observable<MessageInTransit> messages);
 	}
 
 	private final Subject<MessageInTransit> receivedMessages;
 	private final Map<BFTNode, SimulatedNetworkImpl> receivers = new ConcurrentHashMap<>();
-	private final LatencyProvider latencyProvider;
+	private final ChannelCommunication channelCommunication;
 
-	private SimulationNetwork(LatencyProvider latencyProvider) {
-		this.latencyProvider = latencyProvider;
+	@Inject
+	public SimulationNetwork(ChannelCommunication channelCommunication) {
+		this.channelCommunication = Objects.requireNonNull(channelCommunication);
 		this.receivedMessages = ReplaySubject.<MessageInTransit>create(20) // To catch startup timing issues
```

Comment:
I think this should be `createWithSize(20)` — not sure we want to keep all of the messages.